entrypoint.sh

```diff
@@ -1,19 +1,71 @@
 #!/bin/sh -u
+# 3scale (operations@3scale.net)
 
-cd /tmp/
+mkdir /opt/openresty/nginx/lua/
 
-# Download config files
-curl  ${THREESCALE_ENDPOINT}/admin/api/nginx.zip?provider_key=${THREESCALE_PROVIDER_KEY} -o /tmp/nginx.zip
-unzip nginx.zip && rm /tmp/nginx.zip
+reload_openresty() {
+  echo "Reloading Openresty"
+  pkill -USR1 -o nginx
+}
 
-mkdir /opt/openresty/nginx/lua/
+download_threescale_config() {
+  echo "Downloading threescale configuration, using endpoint: ${THREESCALE_ENDPOINT}"
+  cd /tmp
+  rm -rf /tmp/${THREESCALE_PROVIDER_KEY}/
```

Comment:
Imo this is task for `mktemp`. Also, provider key should be treated at least a bit securely, so it should not be part of any path.